src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PrivateWorkerCraftingRequestResolver.java

```diff
@@ -0,0 +1,167 @@
+public class PrivateWorkerCraftingRequestResolver extends AbstractRequestResolver<Stack>
+{
+    public PrivateWorkerCraftingRequestResolver(@NotNull final ILocation location, @NotNull final IToken<?> token)
+    {
+        super(location, token);
+    }
+
+    @Override
+    public TypeToken<? extends Stack> getRequestType()
+    {
+        return TypeToken.of(Stack.class);
+    }
+
+    @Override
+    public boolean canResolve(@NotNull final IRequestManager manager, final IRequest<? extends Stack> requestToCheck)
+    {
+        if (!manager.getColony().getWorld().isRemote)
+        {
+            final Colony colony = (Colony) manager.getColony();
+            final ILocation requesterLocation = requestToCheck.getRequester().getRequesterLocation();
+            final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+            if(building instanceof AbstractBuildingWorker)
+            {
+                final ItemStack stack = requestToCheck.getRequest().getStack();
+                if (ItemStackUtils.isEmpty(stack))
+                {
+                    return false;
+                }
+                return ((AbstractBuildingWorker) building).getFirstRecipe(stack) != null;
+            }
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<IToken<?>> attemptResolve(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends Stack> request)
+    {
+        if (manager.getColony().getWorld().isRemote)
+        {
+            return null;
+        }
+
+        final Colony colony = (Colony) manager.getColony();
+        final ILocation requesterLocation = request.getRequester().getRequesterLocation();
+        final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+        if(canResolve(manager, request) && building instanceof AbstractBuildingWorker)
+        {
+            Log.getLogger().info("Attempt to resolve");
+            final ItemStack stack = request.getRequest().getStack();
+            IRecipeStorage storage = ((AbstractBuildingWorker) building).getFirstFullFillableRecipe(stack);
+            if(storage == null)
+            {
+                storage = ((AbstractBuildingWorker) building).getFirstRecipe(stack);
+                
+                Log.getLogger().info("Request to resolve");
+                final List<IToken<?>> tokens = new ArrayList<>();
+                //todo After simulation has been added, we need to simulate the subrequest and decided wheter to try to resolve it or not.
+                for(final ItemStack neededStack: storage.getInput())
+                {
+                    final Stack stackRequest = new Stack(neededStack);
+                    tokens.add(manager.createRequest(this, stackRequest));
+                }
+
+                return ImmutableList.copyOf(tokens);
+            }
+
+            return Lists.newArrayList();
+        }
+
+        return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1168)